### PR TITLE
Add unit tests for components: copy button + splitter

### PR DESCRIPTION
**Pull request by mrdrogdrog**
### Component/Part
Components splitter and copy button

### Description
This PR adds unit tests for the splitter and copy button component and removes the e2e tests.

### Steps

- [x] Added implementation
- [x] Added / updated tests
- [x] I read the [contribution documentation](https://github.com/hedgedoc/react-client/blob/main/CONTRIBUTING.md) and signed-off my commits to accept the DCO.

### CODE REVIEW

File: src/hooks/common/use-bind-pointer-movement-event-on-window.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+ * @param onPointerMovement is triggered if the user moves the pointer over the window
+ * @param onPointerRelease is triggered if the pointer is released (touch end or mouse up)
+ */
+export const useBindPointerMovementEventOnWindow = (
+  onPointerMovement: (event: MouseEvent | TouchEvent) => void,
+  onPointerRelease: () => void
+) => {
+  useEffect(() => {
+    const pointerMovement = onPointerMovement
+    const pointerRelease = onPointerRelease
```
Comment (DerMolly) [inline src/hooks/common/use-bind-pointer-movement-event-on-window.ts:21]: Why the reassignement?

**Review by DerMolly — CHANGES_REQUESTED** [review]

File: src/hooks/common/use-bind-pointer-movement-event-on-window.ts (hunk 1 above)
Comment (mrdrogdrog) [reply src/hooks/common/use-bind-pointer-movement-event-on-window.ts:21]: This is necessary for the effect. Remove event listener needs EXACTLY the function that was bound before. So i need to save this function here to remove it in the effect return function.

**Review by DerMolly — APPROVED** [review]

File: src/components/editor-page/splitter/splitter.test.tsx [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    Object.defineProperty(window.HTMLDivElement.prototype, 'clientWidth', { value: 1920 })
+    Object.defineProperty(window.HTMLDivElement.prototype, 'offsetLeft', { value: 0 })
+    const view = render(<Splitter showLeft={true} showRight={true} left={<>left</>} right={<>right</>} />)
+    expect(view.container).toMatchSnapshot()
+
+    const divider = await screen.findByTestId('splitter-divider')
+
+    fireEvent.mouseDown(divider, {})
+    fireEvent.mouseMove(window, { buttons: 1, clientX: 1920 })
+    fireEvent.mouseUp(window)
```
Comment (ErikMichelson) [inline src/components/editor-page/splitter/splitter.test.tsx:36]: Maybe also add a test using events of touch gestures?

File: src/components/common/copyable/copy-to-clipboard-button/copy-to-clipboard-button.test.tsx [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+import { CopyToClipboardButton } from './copy-to-clipboard-button'
+import { render, screen } from '@testing-library/react'
+import * as uuidModule from 'uuid'
+
+jest.mock('uuid')
+
+describe('Copy to clipboard button', () => {
+  const copyContent = 'Copy McCopy Content. Electric Copyloo'
+  const copyToClipboardButton = <CopyToClipboardButton content={copyContent} />
+  const uuidMock = 'https://xkcd.com/221/'
```
Comment (ErikMichelson) [inline src/components/common/copyable/copy-to-clipboard-button/copy-to-clipboard-button.test.tsx:23]: Although this seems a funny reference to randomness of UUIDs, I'd rather have some real UUID in here to ensure that characters like the colon or slashes don't have sideeffects (now or in the future).

**Review by ErikMichelson — APPROVED** [review]

**Pull request merged**